packages/core/src/molecule/codec.ts

```diff
@@ -0,0 +1,91 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import { bytesFrom, BytesLike } from "../bytes/index.js";
+import {
+  Num,
+  numBeFromBytes,
+  numBeToBytes,
+  numFromBytes,
+  numToBytes,
+} from "../num/index.js";
+
+export interface Codec<Encodable, Decodable = BytesLike> {
```

Comment:
```typescript
export interface Codec<Encodable, Decoded = Encodable> {
  byteLength?: number;
  encode: (encodable: Encodable) => Bytes;
  decode: (decodable: BytesLike) => Decoded;
}
```

So we can have

```typescript
const TransactionCodec: Codec<ccc.TransactionLike, ccc.Transaction>;
```

How do you think? It seems that the original `Decodable` is never used.

Reply:
good pattern, but I'm afraid it's not possible to explicitly declare the Decoded as ccc.Transaction, because the generic Decoded cannot recognize the `from` method in ccc.Transaction, while we cannot directly use `as` to cast ccc.TransactionLike to ccc.Transaction.

so in case of molecule, I think making the Decoded as BytesLike can fit into most of cases.

Reply:
> good pattern, but I'm afraid it's not possible to explicitly declare the Decoded as ccc.Transaction, because the generic Decoded cannot recognize the `from` method in ccc.Transaction, while we cannot directly use `as` to cast ccc.TransactionLike to ccc.Transaction.
> 
> so in case of molecule, I think making the Decoded as xxxLike can fit into most of cases, for example, users should manually use ccc.Transaction.from to cast the decoded result of molecule decoders

Maybe we can add a method?
```ts
Codec<Encodable, Decoded>.map<NewDecoded>(transformer: (decoded: Decoded) => NewDecoded): Codec<Encodable, NewDecoded>
```